src/main/java/com/connexta/store/controllers/StoreController.java

```diff
@@ -86,24 +85,23 @@
     try {
       inputStream = file.getInputStream();
     } catch (IOException e) {
-      log.warn(
-          "Unable to read file for storeProduct request with for a file with mediaType={} and fileName={}",
-          mediaType,
-          fileName,
+      throw new StoreException(
+          HttpStatus.BAD_REQUEST,
+          String.format(
+              "Unable to read file for storeProduct request with a file with mediaType=%s and fileName=%s",
+              mediaType, fileName),
           e);
-      return ResponseEntity.badRequest().build();
     }
 
     final URI location;
     try {
       location = storeService.createProduct(fileSize, mediaType, fileName, inputStream);
     } catch (StoreException | URISyntaxException e) {
-      log.warn(
-          "Unable to complete storeProduct request for a file with mediaType={} and fileName={}",
-          mediaType,
-          fileName,
+      throw new StoreException(
+          String.format(
+              "Unable to complete storeProduct request for a file with mediaType=%s and fileName=%s",
+              mediaType, fileName),
```

Comment:
✏️ Don't catch StoreException and wrap it in a new StoreException

Reply:
🤦‍♂ good catch